### ISSUE TITLE
Mark laylines do not appear for west longitude, north latitude

### ISSUE THREAD

**Issue by Rasbats** [rgleason/tactics_pi#48]
The same problem as Dashboard_Tactics.
Fixed in this file:
[tactics_pi.cpp.txt](https://github.com/rgleason/tactics_pi/files/13998252/tactics_pi.cpp.txt)

**Comment by rgleason** [comment]
In the version of that you uploaded here tactics.cpp.txt  I find   
```
Search "GCIntersectSegment" (3 hits in 1 file of 1 searched) [Normal]
  C:\Users\fcgle\Downloads\tactics_pi.cpp.txt (3 hits)
	Line 1085: bool GCIntersectSegment(double lat1A, double lon1A, double lat1B, double lon1B,
	Line 1646:           GCIntersectSegment(
	Line 1804:         GCIntersectSegment(pc_end.y, pc_end.x, pc_end2.y, pc_end2.x, pm_end.y,
```

**Comment by rgleason** [comment]
How would I test for this problem, using shipdriver?  I'd like to be sure that it is a problem with this version first.

**Comment by rgleason** [comment]
Mike, I think I downloaded that version and have it. What do I do to test?

**Comment by Rasbats** [comment]
1. Get a grib file for the area you want to test in
2. Install ShipDriver
3. Move your boat to somewhere in the grib area
4. Right-click "Select vessel start position"
5. Press the Start button for ShipDriver
6. The grib file loads and shows the wind where the boat is located
7. Increase speed to say 10kts

If you have tactics_pi loaded you should see the polar and the boat laylines. You may need to sail close hauled by changing course with ShipDriver.

**Comment by Rasbats** [comment]
The version needed is 999 further down the list.

**Comment by rgleason** [comment]
Ah not a VDR file, I was using one of Thomas Rauch's Tactics VDR files. It should be a current grib file, that makes good sense.
I will update to the 999 version now. 
This should be run in the north west?   What about south west?

**Comment by rgleason** [comment]
@Rasbats How do I configure connections to get grib weather data into nmea0183? Are there some connection details needed?

**Comment by Rasbats** [comment]
Oh ... have you set up a !TacticsWP and are tacking towards it?

**Comment by Rasbats** [comment]
The fix works for SouthWest: Here is an example for seas off Chile.

![chile](https://github.com/rgleason/tactics_pi/assets/2537240/81544e67-3bad-4890-b2b5-7befce2d7af8)

**Comment by rgleason** [comment]
Yes, of course I've set a TacticsWP there is no other way to get laylines, but there is no data going into Tactics or into the Nmea0183 window

**Comment by Rasbats** [comment]
I will test on another machine that has not seen this problem.

**Comment by rgleason** [comment]
Now I have an active route on the right because I set the TacticsWP.  Have not seen any data.  The speed is set to be more than 0 but does not show yet.  I click on the Play for Grib too.  

Is there a certain kind of grib needed? Like with very small interval 3hr or 1 hr?
![Screenshot (1893)](https://github.com/rgleason/tactics_pi/assets/2363959/7e5e359f-08db-48a1-8779-9e56e66bc934)

**Comment by Rasbats** [comment]
" I click on the Play for Grib too."
Just starting SD should open the grib file at the correct time. No need to touch the grib_pi apart from selecting the correct grib. ShipDrive is getting the information from the grib using json.

**Comment by Rasbats** [comment]
I am testing Chile with a 6hr grib!

**Comment by rgleason** [comment]
Ok.  That's helpful. I haven't see grib open when SD is opened yet. Will try some more.

I have now first download a grib and loaded it into grib. 
Turn off grib.
1. Move boat here used.
2. Set a TacticsWP  to windward.
3. Hover over TacticsWP and right click and pick "Nav to here" and made another point to go to next to it
4. In any case, the Active Route Console is up.
5. With Grib off, Open Shipdriver hit Start
6. Message Right click and pick ship start location. (Have to first close this message, as can do nothing, not intuitive.)
7. Ok the message, then right click and pick ship start location...  What, it should be "Select Vessel Start Location"? at the bottom. I was not doing that!
8.  But Right click at the vessel start location and then picking "Vessel start location" does not seem to move the vessel.
9. Ok I removed active route, and the tacticsWP.  What order?   
10. I can get it to change headings, but I can't get speed to be >0.  Also gribs does not open automatically.
11.

**Comment by Rasbats** [comment]
It looks like you may not be using SD v999.

**Comment by rgleason** [comment]
Werner would really like this improvement too. He's in Puerto Montt right now, going down through Patagonia next.

**Comment by Rasbats** [comment]
Yes. The same as in Cloudsmith, just the second copy in my Downloads.

**Comment by Rasbats** [comment]
"Ok I removed active route, and the tacticsWP. What order?"
No need to activate a route for tactics_pi. Just drop the tactics mark.

**Comment by rgleason** [comment]
Humm. So I am using the right one.  I am missing something in its operation?  Must I put in an MMSI?

**Comment by rgleason** [comment]
Ok no active route, just drop TacticsWP.

What about "But Right click at the vessel start location and then picking "Vessel start location" does not seem to move the vessel."?

**Comment by rgleason** [comment]
Ok I entered a 9 digit MMSI number and it is starting to work.
I think your MMSI message should say You MUST enter a 9 digit number for this plugin to work.

I can now get speed 6-10knots showing and can control the rudder. 
The own boat does show but it is not moving or turning in accord with shipdriver.

I have changed some grib settings  under playback to. Playback, Loop mode, top of gib, interpolate with 2 min intervals.

I am not getting any movement to the boat and none of the tactics overlays are working for some reason. Is opengl the problem I wonder?
![Screenshot (1898)](https://github.com/rgleason/tactics_pi/assets/2363959/f860ee55-2fe0-4709-9384-fd4cd6117f1f)

**Comment by rgleason** [comment]
@Rasbats  Starting to work.  Have to enter mmsi number.

**Comment by Rasbats** [comment]
"Must I put in an MMSI" - No
"picking "Vessel start location" does not seem to move the vessel." - No, you need to press "Start" after that.
OpenGL has to be on.

**Comment by Rasbats** [comment]
"I am not getting any movement to the boat"
Screenshot is showing movement.

**Comment by rgleason** [comment]
I have my tacticsWP set.  I will set it closer.
OpenGL Accelerated Graphics was never turned off. I checked.
I can turn the boat and steer, I can see the vector for motion, but the boat does not appear to be moving.
I've turned on "Tracking" which helps you see movement and it is moving.

How do I get it obeying the polar selected using the wind from the grib file?.

**Comment by rgleason** [comment]
Really should not make a difference, as this one works in other instances with tactics.
Also the polar file exists here `C:\Users\fcgle\AppData\Local\opencpn\plugins\weather_routing_pi\data\polars\Example\Test-TWS-0-20+60.pol`

but I will use that one....
No improvement.

Ok Closing, SD,  Start VDR with 
[Tactics-sample2-5m.zip](https://github.com/rgleason/tactics_pi/files/14055654/Tactics-sample2-5m.zip)

Leaving the TacticsWP where it is on the other side of the earth....

![Screenshot (1904)](https://github.com/rgleason/tactics_pi/assets/2363959/ef24c63f-0425-402f-962c-d7f6b8458faf)

**Comment by rgleason** [comment]
So what is different about your setup as compared to mine?  It's got to be something.

**Comment by Rasbats** [comment]
I think it is how the grib file is used. Can you post your grib?

**Comment by rgleason** [comment]
@Rasbats  It has to be the order that you activate things, I think.  Is this right?
1. Setup Grib loading the right grib file for the location and time.
2. Setup and Open Tactics with all parameters set for showing polar, laylines, wind feather, etc. (needs Opengl on)
3. place curser for boat, Right click and pick Move boat into start position.
4. place cursor for tactics wp, right click and pick "tactics WP.
5. Now setup to start Shipdriver. Start shipdriver.  
6. At mmsi message, Enter 9 digit mmsi number (fake it if necesary)
7. Hit Start.
8. Move speed to about 10 knots.
9. Move the rudder and zoom into the boat and see it turn.
10. Turn on tracking to see the boat moving.

That is as far as I've gotten. No polar showing, no much data in tactics, very static.

**Comment by rgleason** [comment]
[GFS20240125151801904.grb.docx](https://github.com/rgleason/tactics_pi/files/14055775/GFS20240125151801904.grb.docx)
Remove docx please.

**Comment by Rasbats** [comment]
![japan](https://github.com/rgleason/tactics_pi/assets/2537240/052284f9-26f8-44be-bd21-99faa5cacbb9)

Maybe the grib settings you have. The UTC time of the grib should be automatically selected.

**Comment by Rasbats** [comment]
I recorded this for playing with VDR:
[USA-Wind..txt](https://github.com/rgleason/tactics_pi/files/14056253/USA-Wind.txt)

It shows that mark laylines are not drawn for west longitudes.

**Comment by rgleason** [comment]
Perhaps the problem is the same one you are trying to fix here?  Does not show in wrong hemisphere. I am in Japan.  Note that the current arrow shows.
![Screenshot (1907)](https://github.com/rgleason/tactics_pi/assets/2363959/876f2a0c-4798-47b6-ac7b-7f5397341451)

Well, I finally got it working. I believe it was not Shipdriver, but a Tactics setting that was needed.  Once under Preferences I set Performance Parameters > Force True Wind, it started working.

**Comment by rgleason** [comment]
The problem is that Tactics requires the right data or it does not calculate, So when I am using one of the 3 VDR examples from Thomas that data is all available and "Force true wind is not necessary" and is more inaccurate. Thomas Rauch's instruments were configured for racing and he has a complete set of data.  When we use grib files, true wind has to be derived from other apparent wind using some alternative data for boat speed (GPS) I believe.

**Comment by rgleason** [comment]
This is weird, when I try to setup Japan to run Shipdriver, turn off grib, then select a new grib file for japan. turn off then turn on to check it is right.
Delete the active WP and Move the boat to Japan, and then set a new TacticsWP so everything is ready.
Turn on SD. Hit Start and I get this!!!  I did it twice in a row!
![Screenshot (1912)](https://github.com/rgleason/tactics_pi/assets/2363959/9553b591-c1fa-4c9d-b353-2302f5dff7f7)

**Comment by rgleason** [comment]
Thanks for your help. We've got a phone call coming up at 2.  I'll come back to this.

**Comment by rgleason** [comment]
Mike here is your I recorded this for playing with VDR:
[USA-Wind..txt](https://github.com/rgleason/tactics_pi/files/14056253/USA-Wind.txt)

Results  I hope you can see the laylines are very thin because of my dpi screen. It appears the laylines are missing.
![Screenshot (1915)](https://github.com/rgleason/tactics_pi/assets/2363959/97b252b6-5908-4215-b23a-52911727143c)

**Comment by Rasbats** [comment]
Off for a meal. But yes ... the laylines are not drawn correctly for the mark.

**Comment by Rasbats** [comment]
"Turn on SD. Hit Start and I get this!!! I did it twice in a row! It is supposed to be working in Japan!!"
When you move area you need to use "Select vessel start position" once again. This would have taken the boat back to Japan.

**Comment by rgleason** [comment]
That helped. Using this grib 
[GFS20240125174902161 - Copy.grb.docx](https://github.com/rgleason/tactics_pi/files/14057519/GFS20240125174902161.-.Copy.grb.docx)
In Japan, it appears that the polar and laylines do not work.
![Screenshot (1920)](https://github.com/rgleason/tactics_pi/assets/2363959/4a6f15c1-e44b-4225-84ed-e1cfa2174e9b)

![Screenshot (1919)](https://github.com/rgleason/tactics_pi/assets/2363959/7ea859bf-3804-4b56-a1ff-e8ce3cd65d6c)

**Comment by rgleason** [comment]
That helped. Using this grib 
[GFS20240125174902161 - Copy.grb.docx](https://github.com/rgleason/tactics_pi/files/14057519/GFS20240125174902161.-.Copy.grb.docx)
In Japan, it appears that the polar and laylines do not work.

![Screenshot (1920)](https://github.com/rgleason/tactics_pi/assets/2363959/ccdbb8b1-3d0e-4102-a2f7-b121120faddf)
![Screenshot (1919)](https://github.com/rgleason/tactics_pi/assets/2363959/f38efe3a-ae89-4e07-8cb0-098293d26069)

**Comment by bdbcat** [comment]
Can you show a much simplified test case that shows only the layline geometry problem?

**Comment by Rasbats** [comment]
@bdbcat Tactics_pi line 1348 ff:

```
wxRealPoint intersection_pos;
	intersection_pos = GetLineIntersection(c_end, c_end2, m_end, m_end2);
	if (intersection_pos.x > 0 && intersection_pos.y > 0) {
```
_**intersection_pos.y > 0**_ in west longitudes can never be >0;

**Comment by rgleason** [comment]
Thanks Mike for finding this and being all over it and the good reporting and ShipDriver.
I see the lines after 1348 here https://github.com/rgleason/tactics_pi/blob/master/src/tactics_pi.cpp
I also see the notations and lines after 1376 "Draw the boat laylines, independent from the "Temp. Tactics WP"   https://github.com/Rasbats/tactics_pi/blob/master/src/tactics_pi.cpp#L1376C1-L1376C70

Should these two files be compared?

**Comment by rgleason** [comment]
Comparing these two files using Winmerge and the Compare macro in Notebook++ it appears that the two files have diverged significantly, so much so that it is impossible to determine what has really been changed.
[tactics_pi-rgleason.cpp.txt](https://github.com/rgleason/tactics_pi/files/14066304/tactics_pi-rgleason.cpp.txt)
[tactics_pi-rasbats.cpp.txt](https://github.com/rgleason/tactics_pi/files/14066305/tactics_pi-rasbats.cpp.txt)

It is a fairly sizeable file.
Mike, can you fork my repo to a new project name and move the necessary changes into that, so that I can make a compare and merge those changes?

**Comment by Rasbats** [comment]
@rgleason  Been boat working today. I will fork your repo tomorrow and move the changes.

**Comment by Rasbats** [comment]
PR https://github.com/rgleason/tactics_pi/pull/49 shows my changes. Note that the code for non-intersections remains but is not used.

![pr](https://github.com/rgleason/tactics_pi/assets/2537240/00427326-7b29-47ae-9345-f2c575a120a8)

**Comment by rgleason** [comment]
Thank you MIke!  I am studying this and have a couple of questions. First I tried downloading the tarball and installing it in OpenCPN to see how it worked, but when I try to import it is incompatible and the filename is missing "wx32"
https://cloudsmith.io/~opencpn/repos/tactics-beta/packages/?q=1.0.81

I think I am going to make a new "test" branch and PR your changes to that. Then try to test in OpenCPN, just to be sure.
Thanks. I may be back to you. Hope you're boat work goes well. It's cold and has been raining a whole lot here.

**Comment by rgleason** [comment]
I guess that is the best way to do this.  Made a new branch "laylines" and the made a new PR from rasbats to rgleason/tactics/laylines  in the github web interface.  
https://github.com/rgleason/tactics_pi/tree/laylines

We'll see how it builds and installs.

**Comment by rgleason** [comment]
It built and looks ok  https://cloudsmith.io/~opencpn/repos/tactics-alpha/packages/?q=1.0.81.0+msvc+tarball
It loaded, and I have shipdriver running in Japan towards tacticWP, with grib data going to tactics, and the boat is moving and I can steer it.
There do not appear to be any polar, laylines or current at the boat, also there is no active route line to the tacticWP,  however there is an artifact that disappears when I delete the tacticWP.   Do you have any ideas?
![Screenshot (1934)](https://github.com/rgleason/tactics_pi/assets/2363959/0f91b355-b0f2-47ab-abcb-b5cacc884df5)

![Screenshot (1933)](https://github.com/rgleason/tactics_pi/assets/2363959/20a62578-7a25-409e-aa0d-44bd2adbb185)
![Screenshot (1932)](https://github.com/rgleason/tactics_pi/assets/2363959/c18eff3e-ff67-406b-9e04-76433c9acfa8)

**Comment by Rasbats** [comment]
Easy one ... just close and restart OpenCPN.

**Comment by Rasbats** [comment]
The problem with Japan may be due to the time interval between charts - 6 hours? Try using a grib from the North Sea;
[https://marine.meteoconsult.co.uk/marine-services/grib-files](https://marine.meteoconsult.co.uk/marine-services/grib-files)
These are in 1 hour steps.

**Comment by rgleason** [comment]
Gone back to Adriatic and tried using shipdriver again with this 
[GFS20240125174902161 - Copy.grb.docx](https://github.com/rgleason/tactics_pi/files/14072597/GFS20240125174902161.-.Copy.grb.docx)

Using shipdriver.
When I don't have grib running everything except the polars show laylines etc.
When I turn on the grib file, that all dissappears.
![Screenshot (1935)](https://github.com/rgleason/tactics_pi/assets/2363959/23f8c883-d388-4ea8-ac44-250a745447fa)

![Screenshot (1936)](https://github.com/rgleason/tactics_pi/assets/2363959/ced5301e-86b1-487b-8137-b527152b3a67)

**Comment by rgleason** [comment]
I think you've missed some code.

You should try this version yourself, I think you'll see what I mean.
https://cloudsmith.io/~opencpn/repos/tactics-alpha/packages/?q=1.0.81.0+msvc+tarball

**Comment by Rasbats** [comment]
![adriatic](https://github.com/rgleason/tactics_pi/assets/2537240/bae775fa-00db-45ee-923c-3531dd4efd75)
I needed to alter course to refresh the drawing of the lines.

**Comment by Rasbats** [comment]
To avoid confusion it may be better to use a new version number.

**Comment by rgleason** [comment]
Here is another example. No laylines and no polar even though in adriatic.
![Screenshot (1942)](https://github.com/rgleason/tactics_pi/assets/2363959/35b24a2e-04a0-4456-bc91-f62c3f100909)

Wait, found the problem.... no polar!!

**Comment by rgleason** [comment]
Ok now using Alberg 36 polar. and it is working in the adriatic with the tactics VDR file.
[Tactics-sample3-11m.txt](https://github.com/rgleason/tactics_pi/files/14072761/Tactics-sample3-11m.txt)

![Screenshot (1944)](https://github.com/rgleason/tactics_pi/assets/2363959/fc228e1b-5f1f-4eec-8704-93c0a6b9754a)

There are lots of moving parts.  I guess I have to test it in the other hemisphere next.

**Comment by Rasbats** [comment]
A VDR for the Adriatic, using your grib:
[adriatic.txt](https://github.com/rgleason/tactics_pi/files/14072788/adriatic.txt)

**Comment by rgleason** [comment]
Mike, I went back to Japan, using this grib with it set to interpolate to 30 minutes, and (with a good polar file!) it was showing the laylines and polar file and the tacking angles. However the laylines do not make sense to me. I think some calc might be off.
The laylines don't seem to go to the TacticsWP the head out to wherever. I'll attach the grib and the screenshots.
Then will try your adriatic, but I had that working ok in that part of the globe.

Note: the version being used is in the github/rgleason/tactics/ Laylines branch which has Mikes PR to fix laylines through out the globe.  Deployment is to here in cloudsmith.  https://cloudsmith.io/~opencpn/repos/tactics-alpha/packages/?q=1.0.81.0+msvc+tarball

**Comment by Rasbats** [comment]
My fix fails for Japan. The intersection is completely wrong. Thinking.

**Comment by Rasbats** [comment]
This shows where the intersection should be:

![inter question](https://github.com/rgleason/tactics_pi/assets/2537240/a5ac30bb-cf00-4fbb-99f4-c9255d8cf555)

**Comment by Rasbats** [comment]
At the end of the _GCIntersectSegment_ function I added some corrections.
```
  lat3A *= GEO::RA2DE;
  lon3A *= GEO::RA2DE;
  lat3B *= GEO::RA2DE;
  lon3B *= GEO::RA2DE;

  if (lon1A < -90) {
    double dd = 180 + lon3A;
    dd *= -1;
    lon3A = dd;
  } else if (lon1A > 90) {
    double dd = 180 - lon3A;
    lon3A = dd;  
  }
```
No idea why but this seems to correct the problem.

**Comment by rgleason** [comment]
Very nice fix Mike. I am very impressed and wish I knew enough to do that.  Thanks for all your help fixing this!!!  If you want to PR my repo  laylines  (...I will do that now. and test again)  This is a fast itteration..

**Comment by rgleason** [comment]
Mike your change has been built. It is one of the top two. 
https://cloudsmith.io/~opencpn/repos/tactics-alpha/packages/?q=1.0.81.0+msvc+tarball

I have tested it in Japan, and it appears to be working here a lot better. Have not tested it elsewhere.  Grib file used

[Japan-GFS20240125125301697.grb.pdf](https://github.com/rgleason/tactics_pi/files/14088200/Japan-GFS20240125125301697.grb.pdf)
Remove the pdf.

![Screenshot (1971)](https://github.com/rgleason/tactics_pi/assets/2363959/87cdff44-b278-4c27-9df0-29e42da06bbf)
![Screenshot (1973)](https://github.com/rgleason/tactics_pi/assets/2363959/b884a021-90a7-42a2-b7e8-d95c70cdc54a)
![Screenshot (1974)](https://github.com/rgleason/tactics_pi/assets/2363959/8935f3c2-ad08-4dcd-ae60-37deaa825eb2)
![Screenshot (1975)](https://github.com/rgleason/tactics_pi/assets/2363959/bd115dd8-76be-4ef3-bd29-94d42704c447)

**Comment by rgleason** [comment]
I need to test this some more, in different places on the globe, etc.

**Comment by rgleason** [comment]
Mike, using version Tactics 1.0.81.4  from rgleason/Laylines which has your most recent PR and some TP updates from me.  It appears to me that this version is at least built so that installs, and the filename has "wx32" as it should.
It is using WxWidgets 3.2.2 and it appears to build on all OS. It appears to be installing properly.
The deployment is here  https://cloudsmith.io/~opencpn/repos/tactics-beta/packages/?q=msvc+1.0.81.4+tarball

Tonight I tested 1.0.81.4 it in the Adriatic and Japan, and in both places the laylines seem to have changed, (I think it was better before the last change.) 

I am going to post the results of the two examples. next.

**Comment by rgleason** [comment]
Adriatic
![Screenshot (1984)](https://github.com/rgleason/tactics_pi/assets/2363959/0044b4dd-bd99-4dfc-ad52-05e8b5d113e0)
![Screenshot (1982)](https://github.com/rgleason/tactics_pi/assets/2363959/4e28ef44-cf6a-484d-8f55-87a44ec8ec25)
[Adriatic-GFS20240125174902161 - Copy.grb.pdf](https://github.com/rgleason/tactics_pi/files/14119118/Adriatic-GFS20240125174902161.-.Copy.grb.pdf)


Japan
![Screenshot (1990)](https://github.com/rgleason/tactics_pi/assets/2363959/7d9e25e1-9388-4d6b-997e-40dfbf76b7b2)
![Screenshot (1989)](https://github.com/rgleason/tactics_pi/assets/2363959/c2006211-1840-4820-b81a-e21619cbe67d)

[Japan-GFS20240125125301697.grb.pdf](https://github.com/rgleason/tactics_pi/files/14119119/Japan-GFS20240125125301697.grb.pdf)

**Comment by Rasbats** [comment]
Yes. Using the return value of GCIntersect does not work! Latest PR assumes we always have an intersect.
Version 1.0.81.5
The checks do not create the wx32 version in Cloudsmith. Please can you make this version for testing again.

**Comment by rgleason** [comment]
Mike, if you look at this PR page  https://github.com/rgleason/tactics_pi/pull/55
you will see that for windows wx32 it built properly using the circleci script.  See the green check marks?
I think we will find a good version for wx32 in tactics-alpha, that you created.

Hum, that is another stumper. Its not wx32 but is still wx
https://cloudsmith.io/~opencpn/repos/tactics-alpha/packages/?q=msvc+1.0.81.5+tarball

Going back to your PR build  it looks ok to me 
https://app.circleci.com/pipelines/github/rgleason/tactics_pi/111/workflows/98689a6e-7058-4479-a7d6-f4a08f6b7d5a/jobs/1844
but it is not deploying because of the cloudsmith key.
You are a member of the Cloudsmith Opencpn Plugin Team, I think the permission settings on Tactics are not right. I will fix them.
but I wonder where the wrong version comes from?  Perhaps it is from the Appveyor build that is done and completed from your repo?  Yes, I think that is the answer.

**Comment by rgleason** [comment]
Mike it is found in Tactics-alpha with a search "1.0.81.5+msvc+wx32+tarball"  here  https://cloudsmith.io/~opencpn/repos/tactics-alpha/packages/?q=1.0.81.5+msvc+wx32+tarball

The rest of the macos and debian are completing.  I will try to test it tonight. Thanks so much.
Will look into getting the PR built versions to deploy, but that may be an intentional safeguard.
Don't know why your local to your remote repository pushes don't build the circleci, config.yml script as expected but just build only appveyor?   It may be because you have not enabled circleci Dashboard for Tactics?

**Comment by rgleason** [comment]
Mike version 1.0.81.5 seems to be working for me in Japan, Adriatic and USA east coast.  Should I check in any other locations before merging your PR with master?

Where have you checked it?

![Screenshot (2000)](https://github.com/rgleason/tactics_pi/assets/2363959/2ebc7c91-ed75-4d7b-b389-5856d9f86b73)
![Screenshot (2001)](https://github.com/rgleason/tactics_pi/assets/2363959/f3a445dd-8b33-4e17-9602-2dedf387b1de)
![Screenshot (2002)](https://github.com/rgleason/tactics_pi/assets/2363959/d980aed9-9a29-44e6-8395-e2aa16eb5cd9)

**Comment by Rasbats** [comment]
Tested Vancouver Island, BC and Tasmania. Looks good to me. It would be nice to have other users testing this.
